src/components/hoc/withMeeting.jsx

```diff
@@ -0,0 +1,66 @@
+import React, {useEffect, useContext, useState} from 'react';
+import {concatMap} from 'rxjs/operators';
+import PropTypes from 'prop-types';
+import {AdapterContext} from '../hooks/contexts';
+
+/**
+ * withMeeting is a higher-order component that takes a component (WrappedComponent)
+ * and returns a new component (WithMeeting). The new component needs to be provided with a meeting destination
+ * on the "meetingDestination" prop. The wrapped component will receive the meeting object on the "meeting" prop.
+ *
+ * Example:
+ *
+ * function OriginalComponent {
+ *  return <div>Meeting id is {props.meeting.ID}</div>;
+ *
+ * }
+ *
+ * const EnhancedComponent = withMeeting(OriginalComponent);
+ *
+ * <EnhancedComponent meetingDestination="<MEETING_DESTINATION>" />
+ *
+ * @param {React.Component} WrappedComponent The component to wrap and enhance
+ * @returns {React.Component} The enhanced component
+ */
+export default function withMeeting(WrappedComponent) {
+  /**
+   * @param {string} props.meetingDestination Virtual location where the meeting should take place.
+   * @param props All the props that can be send to the WrappedComponent
+   * @returns {object} JSX of the component
+   */
+  function WithMeeting(props) {
+    const [meeting, setMeeting] = useState({});
+    const {meetingsAdapter} = useContext(AdapterContext);
+
+    useEffect(() => {
+      if (!props.meetingDestination) {
```

Comment:
to have a single return for this useEffect callback, I would suggest the following approach:
```js
let cleanup;
if (!props.meetingDestination) {
  setMeeting({});
  cleanup = undefined;
} else {
  const meetingSubscription = ...
  ...
  cleanup = () => {
    meetingSubscription.unsubscribe();
  };
}

return cleanup;
```